### DISCUSSION TITLE
Unable to get Azure login working

### DISCUSSION THREAD

**Discussion by Azazel666** [outline/outline#3072] [General]
I cannot get azure login to work, i have followed the instructions from https://wiki.generaloutline.com/share/dfa77e56-d4d2-4b51-8ff8-84ea6608faa4

The azure login button shows and when i klick it i get to the microsoft account selector page.
![image](https://user-images.githubusercontent.com/2535654/152761298-c70a7346-43dd-46b7-9e93-ea199b9c4a93.png)

But when i click my account (top one) i just get redirected back to the microsoft account selector again, over and over.

Looking on logs in my docker container for outline wiki there is nothing being logged during login attempts, in crome network tab i only se https://login.microsoftonline.com/common/reprocess? being loaded for each attempt, no loads against the wiki.

Here is a overview screenshot of my app registration in azure
![image](https://user-images.githubusercontent.com/2535654/152761789-e86d4112-5bd2-4116-8ac0-c7a0186c3157.png)

Api permissions
![image](https://user-images.githubusercontent.com/2535654/152761913-d8068952-76e1-46b7-a894-71fba2a80654.png)

And i created a client secret
![image](https://user-images.githubusercontent.com/2535654/152762020-6700deda-37b6-4656-9d37-274ac6c2aa11.png)

And set all this info in the outlook env file as:
```
# To configure Microsoft/Azure auth, you'll need to create an OAuth Client. See
# the guide for details on setting up your Azure App:
# => https://wiki.generaloutline.com/share/dfa77e56-d4d2-4b51-8ff8-84ea6608faa4
AZURE_CLIENT_ID=6ddd12fd-ee6e-40bc-9a0b-#########
AZURE_CLIENT_SECRET=G7M7Q~m4QBFeapBRwff-3ko0fr6########
AZURE_RESOURCE_APP_ID=00000003-0000-0000-c000-000000000000
```

Im using a private azure account/organization and only user exist in the Azure AD.
Any ideas how to trouble shoot this or what i have made wrong in this azure setup?

**Comment by tommoor** [comment]
I have never come across this – it sounds according to this doc like it might be the difference between Azure AD1 vs AD2?

https://stackoverflow.com/questions/44202032/azure-ad-oauth2-redirects-back-to-microsoft-login-after-entering-credentials

**Reply by Azazel666** [reply]
As i understand it im not even using Azure AD1 or AD2 but the Azure AD free.

**Reply by abdou-ghonim** [reply]
@Azazel666 I couldn't even see the provider on the login screen, did you see the provider on the login screen?

**Reply by Shiikos** [reply]
@tommoor Since Microsoft decided to retire AAD1, it is recommended to change the "common" in the authentication address to Azure's directory ID.

**Comment by N4v41** [comment]
the app in azure have a reply url ?

**Comment by matsmagnem** [comment]
I am now experiencing this very same thing when signing in to the Azure Portal using my AAD work account. No custom app registration involved. Also no "common" involved as my tenant id is replacing common. But I also experience this in another app using a custom app registration, where "common" is in the url. So this happens both in "common" and "tenant", and it happens for both custom app registrations and Microsoft's own portals.

**Comment by gnagy** [comment]
I had to add `AZURE_TENANT_ID` to the env file (implemented in this pull request, but missing in docs), and then I got Azure login working.

https://github.com/outline/outline/pull/7028
